payments/models.py

```diff
@@ -39,6 +40,63 @@ def __setattr__(self, key, value):
         return None
 
 
+class BaseSubscription(models.Model):
+    token = models.CharField(
+        _("subscription token/id"),
+        help_text=_("Token/id used to identify subscription by provider"),
+        max_length=255,
+        default=None,
+        null=True,
+        blank=True,
+    )
+    payment_provider = models.CharField(
+        _("payment provider"),
+        help_text=_("Provider variant, that will be used for payment renewal"),
+        max_length=255,
+        default=None,
+        null=True,
+        blank=True,
+    )
+    subscribtion_data = models.JSONField(
+        _("subscription data"),
+        help_text=_("Provider-specific data associated with subscription"),
+        default=dict,
+    )
+
+    class TimeUnit(enum.Enum):
+        year = "year"
+        month = "month"
+        day = "day"
+
+    def set_recurrence(self, token: str, **kwargs):
+        """
+        Sets token and other values associated with subscription recurrence
+        Kwargs can contain provider-specific values
+        """
+        self.token = token
+        self.subscribtion_data = kwargs
+
+    def get_period(self) -> int:
+        raise NotImplementedError
+
+    def get_unit(self) -> TimeUnit:
+        raise NotImplementedError
```

Comment:
What about leaving this up to the subclasses? I mean, I can imagine that this interface might be too flexible for someone (e.g. PayU uses only days so it might be ambiguous how to convert from e.g. years to days) and not enough flexible for others (irregular subscriptions, minute-based subscriptions, day-in-month subscriptions, ...)